.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,124 @@
+  GIT_AUTHOR_EMAIL: ${{ secrets.GIT_COMMIT_AUTHOR_EMAIL_BOT }}
+  GIT_COMMITTER_NAME: ${{ secrets.GIT_COMMIT_USERNAME_BOT }}
+  GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMIT_AUTHOR_EMAIL_BOT }}
+
+jobs:
+  release_prepare:
+    name: Release prepare
+    runs-on: ubuntu-latest
+    # Map the job outputs to step outputs
+    outputs:
+      release_branch: ${{ steps.create_branch_context.outputs.release_branch }}
+      release_ref: ${{ steps.create_context.outputs.release_ref }}
+      tag_ref: ${{ steps.create_context.outputs.tag_ref }}
+
+    steps:
+      # https://github.com/actions/checkout
+      - uses: actions/checkout@v3
+        id: checkout
+        name: checkout master
+        with:
+          ref: ${{ inputs.tag }} # if tag is empty uses the default branch
+
+      # https://github.com/crazy-max/ghaction-import-gpg
+      # https://httgp.com/signing-commits-in-github-actions/
+      - name: Import GPG key
+        id: gpg
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.GPG_KEY_PASSPHRASE_BOT }}
+          passphrase: ${{ secrets.BOT_GPG_KEY_PASSPHRASE }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
+      - name: Remove snapshots suffix
+        id: remove_snapshots
+        run: |
+          mvn versions:set -DremoveSnapshot -DgenerateBackupPoms # remove snapshot and creates a backup of the poms
+
+      - name: get version
+        id: get_version
+        run: |
+          echo "VERSION=$( mvn help:evaluate -Dexpression=project.version -q -DforceStdout )" >> $GITHUB_ENV
+
+      # https://semver.org/
+      - name: Create environment context
+        id: create_context
+        run: |
+          echo "Version is: ${{ env.VERSION }}"
+          if ${{ inputs.is_final_release }}; then
+            echo "is a final release"
+            echo "release_ref=${{ env.VERSION }}" >> $GITHUB_OUTPUT
+            echo "tag_ref=v${{ env.VERSION }}" >> $GITHUB_OUTPUT
+          else
+            echo "is not a final release"
+            echo "release_ref=${{ env.VERSION }}-rc.${{ inputs.candidate_number }}" >> $GITHUB_OUTPUT
+            echo "tag_ref=v${{ env.VERSION }}-rc.${{ inputs.candidate_number }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Create branch context
+        id: create_branch_context
+        run: |
+          echo "release_branch=release/${{ steps.create-context.outputs.release_ref }}" >> $GITHUB_OUTPUT
+
+      - name: create release branch
+        id: create_release_branch
+        run: |
+          echo "Release ref: ${{ steps.create-context.outputs.release_ref }}"
+          git checkout -b release/${{ steps.create-context.outputs.release_ref }}
+
+      - name: Update dependencies
+        id: update_dependencies
+        if: ${{ inputs.set_property_command }} != ''
+        run: |
+          echo "Command to execute: ${{ inputs.set_property_command }}"
+          ${{ inputs.set_property_command }}
```

Comment:
The value of `set_property_command` will be something like: 
`mvn versions:set-property -Dproperty=property.dependency.version -DnewVersion=1.0.0`
or 
`mvn versions:set-property -Dproperty=property.dependency.version, another.dependency.version -DnewVersion=1.0.0`
or 
`mvn versions:set-property -Dproperty=property.dependency.version -DnewVersion=1.0.0 && mvn versions:set-property -Dproperty=another.dependency.version -DnewVersion=1.0.2`
and so on.

It will be the command to run to update a property, in that case used to update the dependency property version to be ready to build the release with the proper release versions.

Reply:
Seems a too generic at the moment, I can pass any unix command in and it will get run (by a privileged user).

Is it currently being used? If not, then I'd drop it for now then once it becomes clearer what is needed then it can be written up in a bash script etc